iceoryx_utils/include/iceoryx_utils/internal/cxx/method_callback.inl

```diff
@@ -0,0 +1,235 @@
+{
+    if (this != &rhs)
+    {
+        m_classPtr = rhs.m_classPtr;
+        m_methodPtr = rhs.m_methodPtr;
+        m_callback = rhs.m_callback;
+
+        rhs.m_classPtr = nullptr;
+    }
+
+    return *this;
+}
+
+template <typename ReturnValue, typename... Args>
+template <typename... MethodArguments>
+inline expected<ReturnValue, MethodCallbackError>
+ConstMethodCallback<ReturnValue, Args...>::operator()(MethodArguments&&... args) const noexcept
+{
+    if (!isValid())
+    {
+        return error<MethodCallbackError>(MethodCallbackError::UNABLE_TO_CALL_METHOD_ON_NULLPTR_CLASS_PTR);
+    }
+
+    return internal::ReturnSuccess<ReturnValue>::call(
+        m_callback, m_classPtr, m_methodPtr, std::forward<MethodArguments>(args)...);
+}
+
+template <typename ReturnValue, typename... Args>
+inline bool ConstMethodCallback<ReturnValue, Args...>::operator==(const ConstMethodCallback& rhs) const noexcept
+{
+    return (m_classPtr == rhs.m_classPtr && m_methodPtr == rhs.m_methodPtr);
+}
+
+template <typename ReturnValue, typename... Args>
+inline bool ConstMethodCallback<ReturnValue, Args...>::operator!=(const ConstMethodCallback& rhs) const noexcept
+{
+    return !(this->operator==(rhs));
+}
+
+template <typename ReturnValue, typename... Args>
+inline bool ConstMethodCallback<ReturnValue, Args...>::isValid() const noexcept
+{
+    return m_classPtr != nullptr;
+}
+
+template <typename ReturnValue, typename... Args>
+inline ConstMethodCallback<ReturnValue, Args...>::operator bool() const noexcept
+{
+    return isValid();
+}
+
+template <typename ReturnValue, typename... Args>
+template <typename ClassType>
+inline void ConstMethodCallback<ReturnValue, Args...>::setObjectPointer(ClassType* const classPtr) noexcept
+{
+    m_classPtr = classPtr;
+}
+
+template <typename ReturnValue, typename... Args>
+template <typename ClassType>
+inline ClassType* ConstMethodCallback<ReturnValue, Args...>::getClassPointer() const noexcept
+{
+    return reinterpret_cast<ClassType*>(m_classPtr);
+}
+
+template <typename ReturnValue, typename... Args>
+template <typename ClassType>
+inline MethodCallback<ReturnValue, Args...>::MethodCallback(ClassType* const classPtr,
+                                                            ReturnValue (ClassType::*methodPtr)(Args...)) noexcept
+    : m_classPtr(classPtr)
+    , m_methodPtr(reinterpret_cast<ReturnValue (internal::GenericClass::*)(Args...)>(methodPtr))
+    , m_callback(internal::methodCallbackCaller<ReturnValue, ClassType, Args...>)
+{
+}
+
+template <typename ReturnValue, typename... Args>
+inline MethodCallback<ReturnValue, Args...>::MethodCallback(MethodCallback&& rhs) noexcept
+{
+    *this = std::move(rhs);
+}
+
+template <typename ReturnValue, typename... Args>
+inline MethodCallback<ReturnValue, Args...>&
+MethodCallback<ReturnValue, Args...>::operator=(MethodCallback&& rhs) noexcept
+{
+    if (this != &rhs)
+    {
+        m_classPtr = rhs.m_classPtr;
+        m_methodPtr = rhs.m_methodPtr;
```

Comment:
In theory for the methodPtr a std::move could make sense, internally it is more than just a pointer (may contain the vtable pointer I think). But I guess it will perform just a (bitwise) copy anyway.

For the others it will not make a difference.

Reply:
fixed